src/Microsoft.Health.Fhir.Api/Controllers/ExportController.cs

```diff
@@ -0,0 +1,136 @@
+         * of operations. Then we can refactor this controller accordingly.
+         */
+
+        private readonly IFhirRequestContextAccessor _fhirRequestContextAccessor;
+        private readonly ExportConfiguration _exportConfig;
+        private readonly ILogger<ExportController> _logger;
+
+        public ExportController(
+            IFhirRequestContextAccessor fhirRequestContextAccessor,
+            IOptions<OperationsConfiguration> operationsConfig,
+            ILogger<ExportController> logger)
+        {
+            EnsureArg.IsNotNull(fhirRequestContextAccessor, nameof(fhirRequestContextAccessor));
+            EnsureArg.IsNotNull(operationsConfig?.Value?.Export, nameof(operationsConfig));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _fhirRequestContextAccessor = fhirRequestContextAccessor;
+            _exportConfig = operationsConfig.Value.Export;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.Export)]
+        [AuditEventType(AuditEventSubType.Export)]
+        public IActionResult Export()
+        {
+            return CheckIfExportIsEnabledAndRespond();
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ExportResourceType)]
+        [AuditEventType(AuditEventSubType.Export)]
+        public IActionResult ExportResourceType(string type)
+        {
+            // Export by ResourceType is supported only for Patient resource type.
+            if (!string.Equals(type, ResourceType.Patient.ToString(), StringComparison.Ordinal))
+            {
+                throw new RequestNotValidException(string.Format(Resources.UnsupportedResourceType, type));
+            }
+
+            return CheckIfExportIsEnabledAndRespond();
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ExportResourceTypeById)]
+        [AuditEventType(AuditEventSubType.Export)]
+        public IActionResult ExportResourceTypeById(string type, string id)
+        {
+            // Export by ResourceTypeId is supported only for Group resource type.
+            if (!string.Equals(type, ResourceType.Group.ToString(), StringComparison.Ordinal) || string.IsNullOrEmpty(id))
+            {
+                throw new RequestNotValidException(string.Format(Resources.UnsupportedResourceType, type));
+            }
+
+            return CheckIfExportIsEnabledAndRespond();
+        }
+
+        /// <summary>
+        /// Currently we don't have any export functionality. We will send the appropriate
+        /// response based on whether export is enabled or not.
+        /// </summary>
+        private FhirResult CheckIfExportIsEnabledAndRespond()
+        {
+            if (!_exportConfig.Enabled)
+            {
+                throw new RequestNotValidException(string.Format(Resources.UnsupportedOperation, "Export"));
+            }
+
+            OperationOutcome result = GenerateOperationOutcome(
```

Comment:
We should probably create something like a `NotSupportedException` and let our normal error handling kick in instead of creating an operation outcome here.

Reply:
Created a `OperationNotImplementException` to handle these cases.

Reply:
We should follow the existing pattern (throw exception here and have the filter convert the except into appropriate OperationOutcome.)